drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -546,8 +551,32 @@ static uint64_t target_time_convert_to_64_bits(uint32_t target_time)
 	return result;
 }
 
-static bool nrf5_tx_at(struct net_pkt *pkt, uint8_t *payload, bool cca)
+static bool nrf5_tx_at(struct nrf5_802154_data *nrf5_radio, struct net_pkt *pkt,
+		   uint8_t *payload, enum ieee802154_tx_mode mode)
 {
+	bool cca = false;
+#if defined(CONFIG_IEEE802154_NRF5_MULTIPLE_CCA)
+	uint8_t extra_cca_attempts = 0;
+#endif
+
+	switch (mode) {
+	case IEEE802154_TX_MODE_TXTIME:
+		break;
+	case IEEE802154_TX_MODE_TXTIME_CCA:
+		cca = true;
+		break;
+#if defined(CONFIG_IEEE802154_NRF5_MULTIPLE_CCA)
```

Comment:
Actually, is there really need to make a separate case for `IEEE802154_EXT_OT_TX_MODE_TXTIME_MULTIPLE_CCA` ?  If `extra_cca_attempts` is zero this will have exactly the same functionality as `IEEE802154_TX_MODE_TXTIME_CCA`.

I don't see really a reason for keeping it as a separate case.

Reply:
It could be handled as you described. However, in my opinion the current implementation provides better separation of concepts, which was the main motivation behind replacing #60343 with this PR:
* `IEEE802154_TX_MODE_TXTIME_CCA` is defined in common namespace as it's a well-defined concept useful in scope of IEEE 802.15.4 spec
* `IEEE802154_EXT_OT_TX_MODE_TXTIME_MULTIPLE_CCA` is an OpenThread-specific extension that is not a part of the common IEEE 802.15.4 interface

I think it's cleaner than handling both behaviors using `IEEE802154_TX_MODE_TXTIME_CCA` since clear separation of common 802.15.4 API and custom extensions is essentially the reason behind this PR

Reply:
Both options would make possible to perform per frame selection of the CCA attempts number just by extending the transmission information metadata of the `otRadioFrame`. Once the metadata needs to be updated anyway, it seems more efficient for me to be able to configure the number of extra CCA attempts rather than single/multiple attempts plus a separated radio API for configuring the number of attempts.

So basically no big preference for any of the two options. But I would tend to go for having `IEEE802154_EXT_OT_TX_MODE_TXTIME_MULTIPLE_CCA` for the reasons @jciupis described.

Reply:
> I think it's cleaner than handling both behaviors using `IEEE802154_TX_MODE_TXTIME_CCA` since clear separation of common 802.15.4 API and custom extensions is essentially the reason behind this PR

+1

@ahasztag In addition to @ankuns striking argument re supporting alternating calls: Having one option only will make the call site conceptually ambiguous (and thereby hard to maintain/read). It's important that we cleanly separate calls that implement IEEE 802.15.4 standard primitives (and document them as such by pointing explicitly to the spec) from calls that implement Thread standard primitives (and document them as well as such) *at the call site*. See https://github.com/zephyrproject-rtos/zephyr/pull/58438#discussion_r1220059230 for a detailed discussion of the rationale behind that approach - TLDR; it's to ensure cross-L2 re-usability - i.e. 15.4 standard primitives will also be used by native L2 while Thread primitives will not. Native L2 expects standard behavior for all calls no matter what the current configuration.

@ankuns I think we need to be careful to always think "API first" (aka *standards first*). We're not implementing low level features and then wrap some API around it. We're developing an API that makes sense from a conceptual/architectural standpoint because it is based (and points to) standard primitives and THEN we provide a low-level driver implementation for it. I mention this explicitly to everyone here without meaning you personally as several recent API additions clearly seem to have originated as simple wrappers around nRF driver features which is what brought us in a difficult situation (again see the link above). So maybe in the future API changes should be discussed in RFCs with separate PRs before an implementation in the nRF driver is attempted - this will considerably reduce the risk of blocked PRs (and thereby actual review/implementation effort) and is a very common approach for API changes in Zephyr AFAIK.